doc/website/blog/2021-09-29-generate-a-fully-working-go-crud-http-api-with-ent.md

```diff
@@ -0,0 +1,550 @@
+            msg := h.stripEntError(err)
+            l.Error(msg, zap.Int("id", id), zap.Error(err))
+            render.BadRequest(w, r, msg)
+        default:
+            l.Error("error fetching pet from db", zap.Int("id", id), zap.Error(err))
+            render.InternalServerError(w, r, nil)
+        }
+        return
+    }
+    d, err := sheriff.Marshal(&sheriff.Options{
+        IncludeEmptyTag: true,
+        Groups:          []string{"pet"},
+    }, e)
+    if err != nil {
+        l.Error("serialization error", zap.Int("id", id), zap.Error(err))
+        render.InternalServerError(w, r, nil)
+        return
+    }
+    l.Info("pet rendered", zap.Int("id", id))
+    render.OK(w, r, d)
+}
+```
+<!--// @formatter:on-->
+
+Next, let’s see how to create an actual RESTful HTTP server that can manage your Pet entities. Create a file
+named `main.go` and add the following content:
+
+```go
+package main
+
+import (
+	"context"
+	"elk-example/ent"
+	elk "elk-example/ent/http"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-playground/validator/v10"
+	_ "github.com/mattn/go-sqlite3"
+	"go.uber.org/zap"
+)
+
+func main() {
+	// Create the ent client.
+	c, err := ent.Open("sqlite3", "./ent.db?_fk=1")
+	if err != nil {
+		log.Fatalf("failed opening connection to sqlite: %v", err)
+	}
+	defer c.Close()
+	// Run the auto migration tool.
+	if err := c.Schema.Create(context.Background()); err != nil {
+		log.Fatalf("failed creating schema resources: %v", err)
+	}
+	// Router, Logger and Validator.
+	r, l, v := chi.NewRouter(), zap.NewExample(), validator.New()
+	// Create the pet handler.
+	r.Route("/pets", func(r chi.Router) {
+		elk.NewPetHandler(c, l, v).Mount(r, elk.PetRoutes)
+	})
+	// Start listen to incoming requests.
+	fmt.Println("Server running")
+	defer fmt.Println("Server stopped")
+	if err := http.ListenAndServe(":8080", r); err != nil {
+		log.Fatal(err)
+	}
+}
+
+```
+
+Next, start the server:
+
+```shell
+go run -mod=mod main.go
+```
+
+Congratulations! We now have a running server serving the Pets API. We could ask the server for a list of all pets in
+the database, but there are none yet. Let’s create one first:
+
+```shell
+curl -X 'POST' -H 'Content-Type: application/json' -d '{"name":"Kuro","age":3}' 'localhost:8080/pets'
+```
+
+You should get this response:
+
+```json
+{
+  "age": 3,
+  "id": 1,
+  "name": "Kuro"
+}
+```
+
+If you head over to the terminal where the server is running you can also see `elk`s built in logging:
+
+```json
+{
+  "level": "info",
+  "msg": "pet rendered",
+  "handler": "PetHandler",
+  "method": "Create",
+  "id": 1
+}
+```
+
+`elk` uses [zap](https://github.com/uber-go/zap) for logging. To learn more about it, have a look at its documentation.
+
+### Relations
+
+To illustrate more of `elk`s features, let’s extend our graph. Edit `ent/schema/user.go` and `ent/schema/pet.go`:
+
+`ent/schema/pet.go`:
+
+<!--// @formatter:off-->
+```go
+// Edges of the Pet.
```

Comment:
Why the `<!--// @formatter:off-->` comment is needed?

```suggestion
```go title="ent/schema/pet.go"
// Edges of the Pet.
```